### PR TITLE
📝 Benchmarking the choice of different network libraries (EPI-26)

### PR DESCRIPTION
**Pull request by 6im0n**
This pull request includes extensive updates to the documentation and benchmarking tests for network libraries. The most important changes include the addition of detailed benchmarking documentation, and the implementation of client-server applications using different network libraries for performance testing.

### Documentation Enhancements:
* [`Documentation/Benchmark_Network.md`](diffhunk://#diff-4fa03527dfa11c6bd673ad63c60679ab1c47ef6424ed40c86b461f8db7144db7R1-R171): Added comprehensive benchmarking documentation, including network definitions, objectives, metrics, and results.

### ASIO Library Tests:
* [`Documentation/tests/ASIO/client.cpp`](diffhunk://#diff-00d0cdd736ceaf114f5361a0576a3e286f20b6d26bb734c1103d3122e4b31a1cR1-R51): Implemented a client application using the ASIO library to measure data transfer speed and latency.
* [`Documentation/tests/ASIO/server.cpp`](diffhunk://#diff-98fa60485be03d73e7cb767a7b92ae3a561e15dd07a697daacd8913c6cb66ebfR1-R58): Implemented a server application using the ASIO library to receive data and measure performance metrics.

### Unix BSD Socket Tests:
* [`Documentation/tests/BSD/client.cpp`](diffhunk://#diff-55a28fda31a61899f6deaf5361c48225b351c1930950a39f57bbc43722839123R1-R73): Added a client implementation using Unix BSD Sockets for benchmarking purposes.
* [`Documentation/tests/BSD/server.cpp`](diffhunk://#diff-7d3e9d44088fadc66de950754e10a96dffd26742bab99d9db4e37a5dac976515R1-R77): Added a server implementation using Unix BSD Sockets to complete the benchmarking setup.

### Winsock Tests:
* [`Documentation/tests/winsocks/CMakeLists.txt`](diffhunk://#diff-e419ff390cf5dcfc681ec1d24647b4b835cbc0bded2ba503777a0ce95d8ce4deR1-R9): Created a CMake configuration for compiling Winsock client and server applications.
* [`Documentation/tests/winsocks/client.cpp`](diffhunk://#diff-e78366fb3302a546753470acddfa506de5cf05346b7d520c6e9909b9b7a3e3faR1-R79): Implemented a client application using Winsock for performance testing on Windows.
* [`Documentation/tests/winsocks/server.cpp`](diffhunk://#diff-28dd285049aeae6dd6c4dfba9f75b021f1678708a61d6d1527a0637e504abfeaR1-R96): Implemented a server application using Winsock to measure data transfer performance.
PS: Tests added for your curiosity, you don't need to understand it

### CODE REVIEW

File: Documentation/Benchmark_Network.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+### Performance Metrics:
+
+| Metric             | ASIO         | Winsock      | Unix BSD Socket |
+|--------------------|--------------|--------------|----------------|
+| **Latency (1KB)**  | 0.04ms       | 0.04ms       | 0.06ms         |
+| **Latency (1MB)**  | 0.2ms        | 0.4ms        | 0.5ms          |
+| **CPU Usage**      | less than 1% | less than 1% | less than 1%   |
+
+---
```
Comment (Marius-P1) [inline Documentation/Benchmark_Network.md]: You can add a text to describe the experiment and the result (what did you do and what result did you observe), then draw a conclusion about which library we'll choose and why

**Review by Marius-P1 — APPROVED** [review]
Great benchmark, just a little part to add and it will be perfect :ok_hand:

File: Documentation/Benchmark_Network.md (hunk 1 above)
Comment (6im0n) [reply Documentation/Benchmark_Network.md]: it's right, i will do it to be more precise for our choice thanks :ok_hand:

File: Documentation/Benchmark_Network.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+- **Winsock (Windows Socket API)**: The primary network interface for Windows systems, optimized for the platform.
+- **Unix BSD Socket (Unix Socket API)**: The standard network interface for Unix-like systems (Linux, macOS, etc.).
+
+**Key Notes**:
+- **ASIO**: A cross-platform library that compiles easily on both Windows and Linux.
+- **Winsock** and **Unix BSD Socket**: Handle networking differently, requiring abstraction layers for full cross-platform compatibility.
+
+---
+
+# Benchmark Results (Placeholder)
```
Comment (RenardFute) [inline Documentation/Benchmark_Network.md:104]: Placeholder ?

File: Documentation/Benchmark_Network.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+### Performance Metrics:
```
Comment (axel-eck) [inline Documentation/Benchmark_Network.md:114]: I might just be off but isn't 'transfer' speed kind of important. I would need some light on this point @6im0n please.

giphy

File: Documentation/Benchmark_Network.md (hunk 2 above)
Comment (axel-eck) [inline Documentation/Benchmark_Network.md:112]: Placeholder ?

**Review by axel-eck — COMMENTED** [review]
I'm not really requesting changes but I would need a bit of informations on the commented things ^^

File: Documentation/Benchmark_Network.md (hunk 2 above)
Comment (6im0n) [reply Documentation/Benchmark_Network.md:112]: Personal note, forget to delete

File: Documentation/Benchmark_Network.md (hunk 3 above)
Comment (6im0n) [reply Documentation/Benchmark_Network.md:114]: Good idea to display it

**Review by AubaneNourry — APPROVED** [review]

**Review by landryarki — APPROVED** [review]

**Pull request merged**